clang/lib/Driver/ToolChains/Gnu.cpp

```diff
@@ -252,9 +252,17 @@ static const char *getLDMOption(const llvm::Triple &T, const ArgList &Args) {
   case llvm::Triple::ppc64le:
     return "elf64lppc";
   case llvm::Triple::riscv32:
-    return "elf32lriscv";
-  case llvm::Triple::riscv64:
-    return "elf64lriscv";
+  case llvm::Triple::riscv64: {
+    bool IsBigEndian = false;
+    if (Arg *A = Args.getLastArg(options::OPT_mlittle_endian,
+                                 options::OPT_mbig_endian))
+      IsBigEndian = A->getOption().matches(options::OPT_mbig_endian);
+
+    if (T.getArch() == llvm::Triple::riscv32)
+      return IsBigEndian ? "elf32briscv" : "elf32lriscv";
+    else
+      return IsBigEndian ? "elf64briscv" : "elf64lriscv";
+  }
```

Comment:
Best as I can tell, this would make RISC-V the only architecture for which there aren't any `ArchType` members distinguishing little-endian vs big-endian? That seems like a strange precedent to set.

Reply:
> Best as I can tell, this would make RISC-V the only architecture for which there aren't any `llvm::Triple::ArchType` members distinguishing little-endian vs big-endian? That seems like a strange precedent to set.

There had been a patch a long time ago to add riscv32be and riscv64be. Not sure if it was in github or if it got left behind in phabricator.

Reply:
I will drop this part for now, since it is dummy/unsed for now. I am planning to create/add support for `BE`, once we are done with these `p8700` patches.
This was a patch on Phabricator: https://reviews.llvm.org/D128612